web_src/js/features/comp/EditorUpload.ts

```diff
@@ -99,7 +99,11 @@ async function handleUploadFiles(editor, dropzoneEl, files, e) {
     const {width, dppx} = await imageInfo(file);
     const placeholder = `[${name}](uploading ...)`;
 
-    editor.insertPlaceholder(placeholder);
+    // Placeholders should be inserted on uploading the thumnail.
+    // If not, the check upload process may fail
+    dropzoneEl.dropzone.on('thumbnail', () => {
+      editor.insertPlaceholder(placeholder);
+    });
```

Comment:
Won't it insert a lot of duplicate event handlers?

Reply:
I revised it to reflect your comment.